src/libs/ajax/metrics/useMetrics.ts

```diff
@@ -0,0 +1,13 @@
+import { useMemo } from 'react'
+import { Ajax } from 'src/libs/ajax'
+
+
+export type CaptureEventFn = (event: string, details?: {}) => void
+
+export const useMetricsEvent = (): CaptureEventFn => {
+  const sendEvent = useMemo(() => Ajax().Metrics.captureEvent, [])
+  const captureEvent: CaptureEventFn = (event: string, details?: {}): void => {
```

Comment:
Why does `sendEvent` need to be wrapped in `captureEvent` instead of returned directly?

Reply:
this avoids some of the Typescript cruft be wrapping the fire-and-forget promise.  Without it Typescript and/or lint complains about the "no explicit void on await" on the consumer side, which is super awkward in some of the nested functional flows to accommodate.  So it's dealt with and wrapped here instead.

Reply:
Oh, I see. Return void instead of Promise<something>. Perhaps worth a comment.

Reply:
good call, ill add a comment